### PR TITLE
Add commit check

### PR DESCRIPTION
**Pull request by CodeChenL**
###### Description of changes
* close https://github.com/radxa-docs/pre-commit-hooks/issues/4
* close https://github.com/radxa-docs/pre-commit-hooks/issues/5

###### Review guide

To review this PR locally, please run the following commands:

```bash
PR= # the current PR number
git fetch origin pull/${PR}/head
git switch --detach FETCH_HEAD
yarn start
yarn start --locale en
# If you want to start 2 locales side-by-side, use the following command on Linux:
# https://github.com/facebook/docusaurus/issues/7377#issuecomment-1167192715
yarn start & DOCUSAURUS_GENERATED_FILES_DIR_NAME=.docusaurus/en yarn start --locale en --port 3001
kill %1
```

### CODE REVIEW

**Comment by CodeChenL** [comment]
效果见https://github.com/CodeChenL/documentation/pull/1

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -52,6 +52,12 @@ jobs:
             EXTRA_ARGS=("--from-ref" "$FROM_REF" "--to-ref" "$TO_REF")
           fi
 
+          if ! pre-commit run --hook-stage push "${EXTRA_ARGS[@]}"
+          then
+            echo "Please check your commit messages"
+            exit 1
+          fi
+
           if [[ ! -z "${{ matrix.test }}" ]]
```
Comment (RadxaYuntian) [inline .github/workflows/deploy.yaml:61]: 我们要求test和test-optional的代码保持同步，所以上面的matrix可以加一个`stage: ["commit", "push", "manual"]`。然后两个测试对应的stage不同

**Review by RadxaYuntian — CHANGES_REQUESTED** [review]

**Review by RadxaYuntian — COMMENTED** [review]
重写历史

**Comment by CodeChenL** [comment]
已压缩commit

**Comment by CodeChenL** [comment]
修复合并冲突又多了一个commit.。。。

**Comment by RadxaYuntian** [comment]
CI挂了：https://github.com/radxa-docs/documentation/actions/runs/8060733327/job/22017250618?pr=604#step:3:199

**Comment by RadxaYuntian** [comment]
怎么会有gitown的ci

**Comment by CodeChenL** [comment]
> 怎么会有gitown的ci

gitown的pr合并了呀

**Comment by CodeChenL** [comment]
这条commit需要压缩不https://github.com/radxa-docs/documentation/pull/604/commits/fa22505629c2695edb963446fc3c9f2048709574

**Comment by RadxaYuntian** [comment]
两个不是一件事就不用

**Review by RadxaYuntian — COMMENTED** [review]
两个commit顺序调换，另外现在第二个commit的类型应该是fix

**Review by RadxaYuntian — APPROVED** [review]

**Pull request merged**